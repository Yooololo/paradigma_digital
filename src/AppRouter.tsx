import { Route, Routes } from "react-router-dom";
import Episode from "./reactComponents/Episode";
import NotFound from "./reactComponents/NotFound";
import Podcast from "./reactComponents/Podcast";
import Podcasts from "./reactComponents/Podcasts";

const AppRouter = () => {
  return (
    <div className="app__container">
      <Routes>
        <Route path="/" element={<Podcasts />} />
        <Route path="/podcast/:podcastId" element={<Podcast />} />
        <Route
          path="/podcast/:podcastId/episode/:episodeId"
          element={<Episode />}
        />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
};

export default AppRouter;
