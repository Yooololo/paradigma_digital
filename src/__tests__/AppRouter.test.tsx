import { render } from "@testing-library/react";
import AppRouter from "../AppRouter";
import { MemoryRouter } from "react-router-dom";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";

describe("AppRouter", () => {
  it('renders the Not Found component when the path is "/*" and not in Router', () => {
    const { container } = render(
      <MemoryRouter initialEntries={[{ pathname: "/asdasd" }]}>
        <IntlProvider locale="en-US" messages={English}>
          <AppRouter />
        </IntlProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll("error")).toBeTruthy();
    expect(container.getElementsByTagName("FormattedMessage")).toBeTruthy();
  });

  it('renders the Episode component when the path is "/podcast/:podcastId/episode/:episodeId"', () => {
    const { container } = render(
      <MemoryRouter
        initialEntries={[{ pathname: "/podcasts/123123/episode/123123" }]}
      >
        <IntlProvider locale="en-US" messages={English}>
          <AppRouter />
        </IntlProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll("podcast__episode")).toBeTruthy();
    expect(container.getElementsByTagName("audio")).toBeTruthy();
  });
});
