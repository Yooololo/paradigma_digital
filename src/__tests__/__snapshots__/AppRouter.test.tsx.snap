// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter renders the Episode component when the path is "/podcast/:podcastId/episode/:episodeId" 1`] = `
<div>
  <div
    class="app__container"
  >
    <div
      class="error dark"
    >
      <p>
        Error 404 - Route Not Found
      </p>
    </div>
  </div>
</div>
`;

exports[`AppRouter renders the Not Found component when the path is "/*" and not in Router 1`] = `
<div>
  <div
    class="app__container"
  >
    <div
      class="error dark"
    >
      <p>
        Error 404 - Route Not Found
      </p>
    </div>
  </div>
</div>
`;
